src/hooks/pokemon.js

```diff
@@ -0,0 +1,55 @@
+import { useEffect, useState } from 'react';
+import {
+  fetchPokemon,
+  fetchPokemonType,
+  fetchTypes,
+  fetchPokemonSearch,
+} from '../services/fetchPokemon.js';
+
+export function usePokemon() {
+  const [pokemon, setPokemon] = useState([]);
+  const [types, setTypes] = useState([]);
+  const [searchTerm, setSearchTerm] = useState('');
+  const [loading, setLoading] = useState(true);
+  // Fetching pokemon names and data from api
+  useEffect(() => {
+    setLoading(true);
+    const fetchData = async () => {
+      const data = await fetchPokemon();
+
+      setPokemon(data);
+      setLoading(false);
+    };
+    fetchData();
+  }, []);
+  // fetching types from API
+  useEffect(() => {
+    const fetchType = async () => {
+      const data = await fetchTypes();
+      setTypes(data);
+    };
+    fetchType();
+  }, []);
+  //
+  const handleTypeChange = async (type) => {
```

Comment:
it'd be better to have this function called inside of a useEffect that triggers every time the type is changed (put type in the dependency array). notice all our other fetching / state setting is occurring inside a useeffect